Added term usage CSV export task, refs #11058

mcantelon opened the pull request:
Added a task to export, as CSV, terms, in a given taxonomy, that have been
assocated with information objects.

jraddaoui commented on lib/flatfile/config/QubitTerm.yml:
```diff
@@ -0,0 +1,9 @@
+  - code
+  - sourceCulture
+  - name
+  - culture
```
Do we really need this file?

jraddaoui commented on lib/task/export/csvExportTermUsageTask.class.php:
```diff
@@ -0,0 +1,139 @@
+
+      return $options['taxonomy-id'];
+    }
+    else if(isset($options['taxonomy-name']))
```
We normally use elseif and it's missing the space after the if.

jraddaoui commented on lib/task/export/csvExportTermUsageTask.class.php:
```diff
@@ -0,0 +1,139 @@
+    3. LEFT JOIN could include unused terms so we could count those too
+    4. Add --mode=used|both|unused-only ("used" as default) to flip between LEFT and INNER join (and maybe add criteria for specific modes)
+
+    */
```
Could you remove the empty lines at the beginning and end and add one after the comment block? And I prefer the full block ...

/*
 *
 *
 */

jraddaoui commented on lib/task/export/csvExportTermUsageTask.class.php:
```diff
@@ -0,0 +1,139 @@
+    4. Add --mode=used|both|unused-only ("used" as default) to flip between LEFT and INNER join (and maybe add criteria for specific modes)
+
+    */
+    $format = 'SELECT DISTINCT t.id FROM %s t INNER JOIN %s r ON r.term_id=t.id WHERE t.taxonomy_id=? ORDER BY t.id';
```
I think we should add another JOIN in this query with the object table, and check the class_name column to find only relations with QubitInformationObject.

jraddaoui reviewed (changes requested):
Looks great Mike! Just a few notes.

jraddaoui commented:
It looks like the commit message refers to a client ticket, could you create a public one explaining the task, etc. And I'm not sure if @fiver-watson would like to add some docs about it.

mcantelon committed:
Added term usage CSV export task, refs #11082

Added a task to export, as CSV, terms, in a given taxonomy, that have been
assocated with information objects.

Made some improvements to QubitFlatfileExport class for this:

* Added ability to set ad-hoc column values (without having to subclass)
* Allow overriding of resource columns (for culture fallback, etc.)
* Add auto-derivation of column names (didn't end up needing this for the
  feature, but it's good to have)

Also changed the bulk base task so core CLI options can be added without
having to add criteria-related options (for cases when a bulk export doesn't
need to support criteria as in this case).

mcantelon commented:
Updated and ready for another review... thanks!

mcantelon commented:
(Oops, didn't mean to close PR.)

MikeFE commented:
LGTM :+1: -- don't forget to make the merge commit msg point to a public ticket when merging into mainline AtoM (can use Redmine's duplicate ticket functionality if one doesn't exist).

mcantelon commented:
Thanks Mike!
